runtime/vm/resolvesupport.cpp

```diff
@@ -1369,17 +1435,71 @@ resolveVirtualMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA c
 			lookupOptions |= J9_LOOK_CLCONSTRAINTS;
 		}
 		
-#if defined(J9VM_OPT_METHOD_HANDLE)
-		/*
-		 * Check for MH.invoke and MH.invokeExact.
-		 *
-		 * Methodrefs corresponding to those methods already have their methodIndex set to index into
-		 * cpClass->methodTypes. We resolve them by calling into MethodType.fromMethodDescriptorString()
-		 * and storing the result into the cpClass->methodTypes table.
-		 */
+
+#if defined(J9VM_OPT_METHOD_HANDLE) || defined(J9VM_OPT_OPENJDK_METHODHANDLE)
 		if ((NULL != cpShapeDescription)
 		&& (resolvedClass == J9VMJAVALANGINVOKEMETHODHANDLE(vm))
 		) {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+			/**
+			 * Check for MH intrinsic methods
+			 *
+			 * Modify the signature to avoid signature mismatch due to varargs
+			 * These methods have special INL send targets
+			 */
+			J9UTF8 *nameUTF = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+			U_8* initialMethodName = J9UTF8_DATA(nameUTF);
+			U_16 initialMethodNameLength = J9UTF8_LENGTH(nameUTF);
+			BOOLEAN isMethodHandle = FALSE;
+
+			switch (initialMethodNameLength) {
+			case 11:
+				if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "invokeBasic")) {
+					isMethodHandle = TRUE;
+				}
+				break;
+			case 12:
+				if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToStatic")) {
+					isMethodHandle = TRUE;
+				}
+				break;
+			case 13:
+				if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToSpecial")
+				||  J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToVirtual")) {
+					isMethodHandle = TRUE;
+				}
+				break;
+			case 15:
+				if (J9UTF8_LITERAL_EQUALS(initialMethodName, initialMethodNameLength, "linkToInterface")) {
+					isMethodHandle = TRUE;
+				}
+				break;
+			}
+
+			if (isMethodHandle) {
+				J9UTF8 *modifiedMethodName = (J9UTF8 *)(nameAndNAS + sizeof(J9ROMNameAndSignature));
+				J9UTF8 *modifiedMethodSig = (J9UTF8 *)(nameAndNAS + sizeof(nameAndNAS) - sizeof(J9UTF8));
+				memset(nameAndNAS, 0, sizeof(nameAndNAS));
```

Comment:
Is the existing nameAndNAS buffer sufficiently large for this?  It would be good to update the comment on that buffer to indicate so:
```
U_8 nameAndNAS[sizeof(J9ROMNameAndSignature) + (sizeof(U_16) + 26 + 5) + sizeof(J9UTF8)];
```